lib/rules/no-dupe-args.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @fileoverview Rule to flag duplicate arguments
+ * @author Jamund Ferguson
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    //--------------------------------------------------------------------------
+    // Helpers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Determines if a given node has duplicate parameters.
+     * @param {ASTNode} node The node to check.
+     * @returns {void}
+     * @private
+     */
+    function checkParams(node) {
+        var currentParam = "";
+        var hasDupArgs = node.params && node.params.map(function(param) {
+            return param.name;
+        }).some(function(param, i, params) {
+            currentParam = param;
+            return params.indexOf(param) !== params.lastIndexOf(param);
+        });
+
+        if (hasDupArgs) {
+            context.report(node, "Duplicate param '{{key}}'.", { key: currentParam});
```

Comment:
I think it's usually fine just to know the line number and the param name (unless you have soooo many params)